ext/node/polyfills/internal/buffer.mjs

```diff
@@ -229,11 +229,22 @@ function fromArrayLike(array) {
   return buf;
 }
 
+function fromUint8Array(u8) {
+  const buf = new Uint8Array(u8.buffer, u8.byteOffset, u8.byteLength);
+  Object.setPrototypeOf(buf, Buffer.prototype);
+  return buf.slice();
```

Comment:
```suggestion
  const buf = new Uint8Array(u8);
  Object.setPrototypeOf(buf, Buffer.prototype);
  return buf;
```

`buf.slice` refers to `Buffer.prototype.slice` which has undesired behavior.

```shellsession
$ cat a.mjs
import { Buffer } from "node:buffer";

const buf = new Uint8Array(1);
Buffer.from(buf)[0] = 1;
console.log(buf[0]);
$ node a.mjs
0
$ deno run a.mjs
0
$ ./deno-canary run a.mjs
1
```

Reply:
@0f-0b would you be willing to open a PR with that fix?

Reply:
> would you be willing to open a PR with that fix?

Opened #24352.